### ISSUE TITLE
JHOVE doesn't handle chained command line flags properly

### ISSUE THREAD

**Issue by carlwilson** [openpreserve/jhove#412]
JHOVE uses custom code for parsing command line parameters that doesn't support chaining flags. For example intuitively `jhove -kr` should enable checksumming and the use of raw data flags but this only happens if `jhove -k -r` is used.

**Comment by marhop** [comment]
If you haven't already settled on something else, I really like [JCommander](http://www.jcommander.org/) for CLI options.

**Comment by tledoux** [comment]
JCommander seems indeed a well-supported library to do that. It has no extra-dependencies, is Apache 2-0 license, requires Java 8.

Another library that appears when searching is [picocli](https://github.com/remkop/picocli): no extra-dependencies, Apache 2.0 license, works with Java 5. The [comparison ](https://github.com/remkop/picocli/wiki/picocli-vs-JCommander), made by the author itself, insists in the fact that it can be integrated as source (no dependency at all) and it Supports for clustered POSIX short options, which is what this issue is about...

**Comment by marhop** [comment]
Oops, you're right! I thought JCommander supported clustered options, but I haven't checked in a while (seems it does not). Indeed picocli looks nice, and it even supports autocomplete.

**Comment by carlwilson** [comment]
Hi, @DinoAGW is this the right issue to assign you to, or more accurately is this your GitHub id? I'm happy to help move this along or write some unit tests. It's a bit cheeky but if we're going to replace this with picoli then we might look at #413 as well which is related.

**Comment by DinoAGW** [comment]
@carlwilson yes, I am. That would be great. I'm looking forward to start with making myself familiar with picoli.

**Comment by carlwilson** [comment]
That's great @DinoAGW, I'll likely not start until tomorrow but will get a few tests written. There's some implied refactoring here as the code's not testable in its current state. It'll need a dedicated parsing class that returns some kind of config instance, at least in my head. I might need to do that as I write the tests.

**Comment by MartinSpeller** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [JHOVE doesn't handle chained command line flags properly #412 - Assigned to DinoAGW](https://trello.com/c/AJBtM0T1/52-jhove-doesnt-handle-chained-command-line-flags-properly-412-assigned-to-dinoagw)

**Comment by DinoAGW** [comment]
[here](https://github.com/DinoAGW/jhove) is a first allmost complete try to solve it.
There are a few //TODO and "tbd"s in there because I don't know Jhove, Java and englisch well enough to make everything professional by myself, but I think, it is ready to get testet.
